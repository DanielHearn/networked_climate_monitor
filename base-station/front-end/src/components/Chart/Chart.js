import { Line, mixins } from 'vue-chartjs'
const { reactiveProp } = mixins

// Displays a line chart
export default {
  extends: Line,
  mixins: [reactiveProp],
  props: ['options'],
  mounted() {
    // Render chart using chart data and options bound to component
    this.renderChart(this.chartData, this.options)
  }
}
