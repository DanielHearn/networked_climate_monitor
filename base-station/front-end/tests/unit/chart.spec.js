import { mount } from '@vue/test-utils'
import Chart from './../../src/components/Chart/Chart.js'

describe('Chart.vue', () => {
  let wrapper

  beforeEach(() => {
    wrapper = mount(Chart, {
      propsData: {
        chartData: {
          labels: ['January', 'February'],
          datasets: [
            {
              label: 'Data One',
              backgroundColor: '#f87979',
              data: [40, 20]
            }
          ]
        },
        options: {
          responsive: true,
          maintainAspectRatio: false
        }
      }
    })
  })

  afterEach(() => {
    wrapper.destroy()
  })

  it('props', () => {
    expect(wrapper.find('#line-chart').exists()).toBe(true)
  })

  it('has the expected html structure', () => {
    expect(wrapper.element).toMatchSnapshot()
  })
})
