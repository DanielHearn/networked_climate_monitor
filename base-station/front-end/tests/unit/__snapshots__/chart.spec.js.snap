// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chart.vue has the expected html structure 1`] = `
<div
  class=""
>
  <canvas
    class="chartjs-render-monitor"
    height="0"
    id="line-chart"
    style="display: block;"
    width="0"
  />
</div>
`;
